testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestSparkCompatibility.java

```diff
@@ -0,0 +1,175 @@
+
+        QueryResult prestoSelect = onPresto().executeQuery(format("%s FROM %s", startOfSelect, prestoTableName(baseTableName)));
+        assertThat(prestoSelect).containsOnly(row1, row2, row3, row4);
+
+        onSpark().executeQuery("DROP TABLE " + baseTableName);
+    }
+
+    @Test(groups = {HIVE_SPARK_BUCKETING, PROFILE_SPECIFIC_TESTS})
+    public void testPrestoReadingTableCreatedByHiveCompliantSpark()
+    {
+        String baseTableName = "test_presto_reading_spark_buckets";
+
+        String sparkTableDefinition =
+                "CREATE TABLE %s (" +
+                        "  string_col STRING" +
+                        ", _bigint BIGINT" +
+                        ", _integer INTEGER" +
+                        ", _real REAL" +
+                        ", _double DOUBLE" +
+                        ", _boolean BOOLEAN" +
+                        ") USING HIVE OPTIONS (fileFormat 'ORC')";
+        onSpark().executeQuery(format(sparkTableDefinition, baseTableName));
+
+        onHive().executeQuery(format("ALTER TABLE %s CLUSTERED BY (string_col) INTO 4 BUCKETS", baseTableName));
+
+        String values = "VALUES " +
+                "('one', 1000000000000000, 1000000000, 10000000.123, 100000000000.123, true)" +
+                ", ('two', -1000000000000000, -1000000000, -10000000.123, -100000000000.123, false)" +
+                ", ('three', 2000000000000000, 2000000000, 20000000.123, 200000000000.123, true)" +
+                ", ('four', -2000000000000000, -2000000000, -20000000.123, -200000000000.123, false)";
+        String insert = format("INSERT INTO %s %s", baseTableName, values);
+        onSpark().executeQuery(insert);
+
+        Row row1 = row("one", 1000000000000000L, 1000000000, 10000000.123F, 100000000000.123, true);
+        Row row2 = row("two", -1000000000000000L, -1000000000, -10000000.123F, -100000000000.123, false);
+        Row row3 = row("three", 2000000000000000L, 2000000000, 20000000.123F, 200000000000.123, true);
+        Row row4 = row("four", -2000000000000000L, -2000000000, -20000000.123F, -200000000000.123, false);
+
+        String startOfSelect = "SELECT string_col, _bigint, _integer, _real, _double, _boolean";
+        QueryResult sparkSelect = onSpark().executeQuery(format("%s FROM %s", startOfSelect, baseTableName));
+        assertThat(sparkSelect).containsOnly(row1, row2, row3, row4);
+
+        assertThat(() -> onPresto().executeQuery(format("%s FROM %s", startOfSelect, prestoTableName(baseTableName))))
+                .failsWithMessage("Hive table is corrupt.");
+
+        // Spark only sets provider in native mode, we have to do this manually
+        onHive().executeQuery(format("ALTER TABLE %s SET TBLPROPERTIES ('%s' = 'hive')", baseTableName, SPARK_TABLE_PROVIDER_KEY));
+
+        QueryResult prestoSelect = onPresto().executeQuery(format("%s FROM %s", startOfSelect, prestoTableName(baseTableName)));
+        assertThat(prestoSelect).containsOnly(row1, row2, row3, row4);
```

Comment:
Add a test that selecting `$bucket` or filtering on `$bucket = 0` fail with column not found